packages/peregrine/lib/talons/OrderHistoryPage/useOrderHistoryPage.js

```diff
@@ -24,76 +24,69 @@ export const useOrderHistoryPage = (props = {}) => {
-    const formState = useFormState();
-    const formApi = useFormApi();
 
-    const searchText = useMemo(() => {
-        return formState.values.search;
-    }, [formState]);
+    const [pageSize, setPageSize] = useState(PAGE_SIZE);
+    const [searchText, setSearchText] = useState('');
 
-    const [
-        getOrderData,
-        { loading: orderLoading, error: getOrderError }
-    ] = useLazyQuery(getCustomerOrderQuery, {
+    const {
+        data: orderData,
+        error: getOrderError,
+        loading: orderLoading
+    } = useQuery(getCustomerOrdersQuery, {
         fetchPolicy: 'cache-and-network',
-        nextFetchPolicy: 'cache-first',
-        onCompleted: orderData => {
-            setOrders(orderData.customer.orders.items);
+        variables: {
+            filter: {
+                number: {
+                    match: searchText
+                }
+            },
+            pageSize
         }
     });
 
-    const getOrdersData = useCallback(() => {
-        getOrderData({
-            variables: {
-                orderNumber: {
-                    number: {
-                        match: ''
-                    }
-                }
-            }
-        });
-    }, [getOrderData]);
+    const orders = orderData ? orderData.customer.orders.items : [];
+
+    const isLoadingWithoutData = !orderData && orderLoading;
+    const isBackgroundLoading = !!orderData && orderLoading;
+
+    const pageInfo = useMemo(() => {
+        if (orderData) {
+            const { total_count } = orderData.customer.orders;
+
+            return {
+                current: pageSize < total_count ? pageSize : total_count,
+                total: total_count
+            };
+        }
 
-    const isLoadingWithoutData = !orders && orderLoading;
-    const isBackgroundLoading = !!orders && orderLoading;
+        return null;
+    }, [orderData, pageSize]);
 
     const derivedErrorMessage = useMemo(
         () => deriveErrorMessage([getOrderError]),
         [getOrderError]
     );
 
-    const getOrderDetails = useCallback(() => {
-        if (!isBackgroundLoading) {
-            getOrderData({
-                variables: {
-                    orderNumber: {
-                        number: {
-                            match: searchText
-                        }
-                    }
-                }
-            });
-        }
-    }, [isBackgroundLoading, getOrderData, searchText]);
+    const handleReset = useCallback(() => {
+        setSearchText('');
+    }, []);
 
-    const resetForm = useCallback(
-        event => {
-            event.stopPropagation();
+    const handleSubmit = useCallback(({ search }) => {
+        setSearchText(search);
+    }, []);
 
-            formApi.reset();
-            getOrdersData();
-        },
-        [formApi, getOrdersData]
-    );
+    const loadMoreOrders = useMemo(() => {
+        if (orderData) {
+            const { page_info } = orderData.customer.orders;
```

Comment:
I'm a little fuzzy on the difference between this `page_info` and the one locally defined (L52).

Reply:
The local defined feeds the count indicator `Showing {x} of {y}` and the `page_info` data response is just used to determine if we're on the last page or not. The order this was developed in isn't obvious, but the count indicator came in last minute, and I suppose this `page_info` isn't really needed since we can derive this by `pageSize < total_count`. Multiple ways to derive, but happy to clean this up if you think it will make it easier to understand.